lib/api/CDataFrameAnalysisSpecification.cc

```diff
@@ -100,11 +96,14 @@ CDataFrameAnalysisSpecification::CDataFrameAnalysisSpecification(TRunnerFactoryU
         m_NumberColumns = parameters[COLS].as<std::size_t>();
         m_MemoryLimit = parameters[MEMORY_LIMIT].as<std::size_t>();
         m_NumberThreads = parameters[THREADS].as<std::size_t>();
-        m_TemporaryDirectory = parameters[TEMPORARY_DIRECTORY].fallback(
-            boost::filesystem::current_path().string());
+        m_TemporaryDirectory = parameters[TEMPORARY_DIRECTORY].fallback(std::string{});
         m_ResultsField = parameters[RESULTS_FIELD].fallback(DEFAULT_RESULT_FIELD);
         m_DiskUsageAllowed = parameters[DISK_USAGE_ALLOWED].fallback(DEFAULT_DISK_USAGE_ALLOWED);
 
+        if (m_DiskUsageAllowed and m_TemporaryDirectory.empty()) {
+            HANDLE_FATAL(<< "Temporary directory path should be explicitly set if disk usage is allowed!");
+        }
```

Comment:
I've been trying to be systematic in visible errors. As such I'd consider changing to
```
HANDLE_FATAL("Input error: temporary directory path should be explicitly set if disk"
             " usage is allowed! Please report this problem.");
```

Reply:
done